### PR TITLE
testing/telegram-tdlib: new aport

### PR DESCRIPTION
**Pull request by OTLabs**
https://core.telegram.org/tdlib
Cross-platform library for building Telegram clients

### CODE REVIEW

File: testing/telegram-tdlib/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# Maintainer: Oleg Titov <oleg.titov@gmail.com>
+pkgname=telegram-tdlib
+pkgver=1.4.0
+pkgrel=0
+pkgdesc="Cross-platform library for building Telegram clients"
+url="https://core.telegram.org/tdlib"
+arch="all"
+license="BSL-1.0"
+options="!check" # No test suite from upstream
+depends=""
```
Comment (maxice8) [inline testing/telegram-tdlib/APKBUILD]: empty variables can be removed for brevity

File: testing/telegram-tdlib/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+makedepends="cmake openssl-dev zlib-dev gperf linux-headers"
+subpackages="$pkgname-static $pkgname-dev"
+source="$pkgname-$pkgver.tar.gz::https://github.com/tdlib/td/archive/v$pkgver.tar.gz"
+builddir="$srcdir/td-$pkgver"
+
+build() {
+	mkdir build
+	cd build
+	cmake -DCMAKE_BUILD_TYPE=Release \
```
Comment (maxice8) [inline testing/telegram-tdlib/APKBUILD]: i think

```cmake
cmake .. \
  -DOtherArgs= \
  -DFitBetter= \
```

File: testing/telegram-tdlib/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		-DCMAKE_INSTALL_PREFIX:PATH="$pkgdir/usr" \
+		..
+	cmake --build .
+}
+
+package() {
+	cd build
+	mkdir -p ${pkgdir}/usr
+	cmake --build . --target install
```
Comment (maxice8) [inline testing/telegram-tdlib/APKBUILD]: i prefer `make install DESTDIR="$pkgdir"`

File: testing/telegram-tdlib/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+source="$pkgname-$pkgver.tar.gz::https://github.com/tdlib/td/archive/v$pkgver.tar.gz"
+builddir="$srcdir/td-$pkgver"
+
+build() {
+	mkdir build
+	cd build
+	cmake -DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_INSTALL_PREFIX:PATH="$pkgdir/usr" \
+		..
+	cmake --build .
```
Comment (maxice8) [inline testing/telegram-tdlib/APKBUILD]: i prefer `make`

File: testing/telegram-tdlib/APKBUILD (hunk 1 above)
Comment (OTLabs) [reply testing/telegram-tdlib/APKBUILD]: Fixed. Thank you! Really missed it, it supposed I already know that :-)

File: testing/telegram-tdlib/APKBUILD (hunk 2 above)
Comment (OTLabs) [reply testing/telegram-tdlib/APKBUILD]: Fixed. Thank you!

File: testing/telegram-tdlib/APKBUILD (hunk 4 above)
Comment (OTLabs) [reply testing/telegram-tdlib/APKBUILD]: Fixed. Thank you!

File: testing/telegram-tdlib/APKBUILD (hunk 3 above)
Comment (OTLabs) [reply testing/telegram-tdlib/APKBUILD]: Fixed. Thank you!

**Comment by Ikke** [comment]
The code could even be more simplified:

```diff
diff --git a/testing/telegram-tdlib/APKBUILD b/testing/telegram-tdlib/APKBUILD
index 581da4a7be..d35190ded6 100644
--- a/testing/telegram-tdlib/APKBUILD
+++ b/testing/telegram-tdlib/APKBUILD
@@ -16,23 +16,15 @@ builddir="$srcdir/td-$pkgver"
 build() {
        mkdir build

-       cd build
-
-       cmake .. \
+       cmake -B build \
                -DCMAKE_BUILD_TYPE=Release \
                -DCMAKE_INSTALL_PREFIX:PATH="$pkgdir/usr"

-       make
+       make -C build
 }

 package() {
-       cd build
-
-       mkdir -p ${pkgdir}/usr
-#      make install DESTDIR="$pkgdir"
-       make install
-
-       cd "$builddir"
+       make -C build install

        install -m644 -D -t "$pkgdir/usr/share/licenses/$pkgname" \
                LICENSE_1_0.txt
```

Admitatly, the changes in the build function are only marginally better, so they could be skipped, but the package function can be done simpler.

**Comment by andypost** [comment]
It fails on s390x https://build.alpinelinux.org/buildlogs/build-edge-s390x/testing/telegram-tdlib/telegram-tdlib-1.4.0-r0.log
Guess better to disable

**Comment by OTLabs** [comment]
I will submit new PR to disable s390x.

The error is:
```
Unsupported tl-schema version -1
make[2]: *** [td/generate/CMakeFiles/tl_generate_common.dir/build.make:63: td/generate/CMakeFiles/tl_generate_common] Aborted
make[2]: Leaving directory '/home/buildozer/aports/testing/telegram-tdlib/src/td-1.4.0/build'
make[1]: *** [CMakeFiles/Makefile2:781: td/generate/CMakeFiles/tl_generate_common.dir/all] Error 2
```

**Comment by OTLabs** [comment]
Disabled arch s390x in https://github.com/alpinelinux/aports/pull/7711

**Pull request closed without merging**